ide/editor.lib/src/org/netbeans/lib/editor/hyperlink/spi/HyperlinkProviderExt.java

```diff
@@ -125,5 +126,17 @@
      * @since 1.18
      */
     String getTooltipText(Document doc, int offset, HyperlinkType type);
-    
+
+    /**
+     * Returns hyperlink target location.
+     *
+     * @param doc document on which to operate.
+     * @param offset &gt;=0 offset to test (it generally should be offset &lt; doc.getLength(), but
+     *               the implementations should not depend on it)
+     * @param type the hyperlink type
+     * @return 4.20
+     */
+    default CompletableFuture<HyperlinkLocation> getHyperlinkLocation(Document doc, int offset, HyperlinkType type) {
```

Comment:
Adding default method is an option, but we have added new `CompletionCollector` interface. Shoudn't we rather add new interface `HyperlinkLocation.Provider extends HyperlinkProviderExt` as well?

Reply:
Fixed. New interface created in `api.lsp` module.